src/status_im/contexts/wallet/account/tabs/about/view.cljs

```diff
@@ -70,7 +70,17 @@
         {:keys [address path watch-only?]}        (rf/sub [:wallet/current-viewing-account])
         {keypair-name :name
          keypair-type :type}                      (rf/sub [:wallet/current-viewing-account-keypair])
-        networks                                  (rf/sub [:wallet/network-preference-details])]
+        networks                                  (rf/sub [:wallet/network-preference-details])
+        origin-type                               (rn/use-memo
+                                                   #(case keypair-type
+                                                      :seed
+                                                      :recovery-phrase
+
+                                                      :key
+                                                      :private-key
+
+                                                      :default-keypair)
+                                                   [keypair-type])]
```

Comment:
Instead of using `use-memo` we can define this function outside the component. Something like:

```clojure
(defn my-fn [keypair-type]
  (case leypair-type
    :a :b 
    :c :d
    :e))
```

By doing this, the funciton becomes pure and the logic  inside the component gets simpler

Reply:
The problem is not caching this `case` call, the problem is when we redefine this function (bound to `origin-type`) and it triggers undesiderd re-renders, so the answer to:
 
 > Why would we cache such a simple expression?
 
is: Because it's a function, and it may cause re-renders when it's passed to children components.
We could just move this funciton to a `defn` so we keep the reference stable.

Reply:
@ulisesmac, I don't quite understand your last points. I think we're talking about the same thing in the end :) This `case` in the PR is a macro and it's processed even faster than a single `get` call in Clojure or also faster than a destructure in a let binding, which we do all the time without thinking. Thus, this `case` can be recomputed every single time when the view renders, no problem. That's why I mentioned there's no point in caching this or even worrying about extracting to a function var (unless the dev wants to do this for code readability).

Reply:
@ilmotta  oh yeah, I agree!

I completely confused the `use-memo` with `use-callback` :man_facepalming: 

I agree with you, memoizing this seems more expensive than just calculating it :+1:

Reply:
Thank you @ulisesmac & @ilmotta for the informative discussion about the `case` macro 🙏 .

I will remove the wrapped `use-memo` 👍